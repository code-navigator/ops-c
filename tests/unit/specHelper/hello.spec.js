import Vuetify from 'vuetify'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import test from './../../../src/components/controls/hello'

describe('test.vue', () => {
  let localVue

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(Vuetify)
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('should include a message', () => {
    const wrapper = shallowMount(test)
    expect(wrapper.html()).toContain('Hello Mark')
  })

  it('should render correctly', () => {
    const wrapper = shallowMount(test)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should trigger click event', () => {
    const onClick = jest.fn()
    const wrapper = shallowMount(test,
      {
        propsData: {onClick}
      })
    wrapper.find('button').trigger('click')
    let output = wrapper.find('#output').element.textContent
    expect(output).toBe('1')
    wrapper.find('button').trigger('click')
    // output = wrapper.find('#output').element.textContent
    expect(wrapper.vm.number).toBe(2)
  })
})
