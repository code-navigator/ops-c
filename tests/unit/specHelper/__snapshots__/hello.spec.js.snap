// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.vue should render correctly 1`] = `
<div>
  <h1>Hello Mark</h1>
  <button></button>
  <p id="output">0</p>
</div>
`;
